tests/test_default_values.py

```diff
@@ -35,3 +37,72 @@ def test_default_values(self):
         self.assertEqual(config_suite.snapshot.color, {})
         self.assertEqual(config_suite.snapshot.minor_incidents, [])
 
+    def test_default_presedence(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        raw_config["tire"] = {}
+        raw_config["tire"]["dimension"] = 15
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+        self.assertEqual(config_suite.snapshot.tire.rim, "green")
+        self.assertEqual(config_suite.snapshot.tire.dimension, 15)
+
+    def test_partial_default_values(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        car_schema[MK.Content]["tire"][MK.Content]["dimension"].pop("default")
+
+        raw_config["tire"] = {}
+        raw_config["tire"]["dimension"] = 15
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+        self.assertEqual(config_suite.snapshot.tire.rim, "green")
+        self.assertEqual(config_suite.snapshot.tire.dimension, 15)
+
+    def test_invalid_defaults(self):
+        car_schema = car.build_schema()
+        raw_config = car.build_all_default_config()
+        configsuite.ConfigSuite(raw_config, car_schema)
+
+        car_schema[MK.Content]["tire"][MK.Content]["dimension"][
+            MK.Default
+        ] = "incorrect default type"
+        with self.assertRaises(ValueError):
+            configsuite.ConfigSuite(raw_config, car_schema)
+            # self.assertTrue(msg.exception.startswidth("default problems!"))
```

Comment:
Would like to assert the message to be sure it's the correct exception that is caught and not another `ValueError` somewhere else. `unittest2` has this functionality, but unfortunately not `unittest`...

Reply:
Perhaps not as nice, but:

```python
with self.assertRaises(ValueError) as error_context:
    configsuite.ConfigSuite(raw_config, car_schema)

self.assertTrue(error_context.exception.startswidth("default problems!"))
```
should work... We can of course also consider dragging in `unittest2`, but I would assume that this will do for now?

Reply:
This is how it's done in `unittest2` as well. I tried that, somehow I managed to mess it up.. Apparently it's backported to unittest, so I'll remove the comments and have them execute!

Reply:
```
        with self.assertRaises(ValueError) as msg:
            configsuite.ConfigSuite(raw_config, car_schema)
>       self.assertTrue(msg.exception.startswidth("default problems!"))

E       AttributeError: 'exceptions.ValueError' object has no attribute 'startswidth'
```
Yeh.. this is what I tested the first time.. am I missing something?

Reply:
Right.. Damn it! alright... I'll fix `str` it!